src/connector/src/test_data/proto_recursive/recursive.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package recursive;
+
+message ComplexRecursiveMessage {
+  string node_name = 1;
+  int32 node_id = 2;
+
+  message Attributes {
+    string key = 1;
+    string value = 2;
+  }
+
+  repeated Attributes attributes = 3;
+
+  message Parent {
+    string parent_name = 1;
+    int32 parent_id = 2;
+    repeated ComplexRecursiveMessage siblings = 3;
+  }
+
+  Parent parent = 4;
```

Comment:
Is this a practical example? Seems we're storing the nodes multiple times since protobuf is by-value instead of by-ref. 😂

Reply:
No, the example is generated by chatgpt. real world ones are welcomed